stream-legal-entity/legal-entity-core/src/main/java/com/backbase/stream/ServiceAgreementSagaV2.java

```diff
@@ -146,7 +152,29 @@ public Mono<ServiceAgreementTaskV2> executeTask(@SpanTag(value = "streamTask") S
             .flatMap(this::setupAdministratorPermissions)
             .flatMap(this::setupLimits)
             .flatMap(this::processProducts)
-            .flatMap(this::postContacts);
+            .flatMap(this::postContacts)
+            .flatMap(this::updatePlans);
+    }
+
+    private Mono<ServiceAgreementTaskV2> updatePlans(ServiceAgreementTaskV2 streamTask){
+        ServiceAgreementV2 serviceAgreement = streamTask.getServiceAgreement();
+        if (!plansSaga.isEnabled()) {
+            streamTask.info(SERVICE_AGREEMENT, PROCESS_PLANS, SKIPPED, serviceAgreement.getExternalId(), serviceAgreement.getInternalId(),
+                    "Plan Saga configured to skipped");
+            return Mono.just(streamTask);
+        }
+        log.info("Updating Plan for Service Agreement Id {}", serviceAgreement.getExternalId());
+        serviceAgreement.getJobProfileUsers().stream().forEach(jobProfileUser -> {
```

Comment:
Does it make sense to update all the Users of a service agreement to another plan? Users in a service agreement can be the primary user associated with a _Customer_ but also their accountant. I would assume the Plan is with the customer, not the user.